### ISSUE TITLE
Feature - Need to add GPU/CPU switch detection logic because not all devices can be assumed to have GPUs for setup!

### ISSUE THREAD

**Issue by ghost** [ac-i2i-engineering/matching-engine#11]
1. Request to add the following virtual environment setup for cleaner installation and containerization before proceeding to `pip install`:

```
cd /path/to/your/repo
python3 -m venv env
source env/bin/activate
```

2. Currently facing this issue during `pip install -r requirements.txt` because my Mac doesn't have CUDA/GPU support. Can you please add a switch-case that detects if there is a GPU, and if not relies on CPU for the computation. In the worst case, let's have the main branch contain the CPU defaulting code, and you can specify an argument / switch to a new branch `master-gpu`, where we can have the GPU code?

![image](https://github.com/user-attachments/assets/661acd14-57e8-4e73-bcb0-25f24537d0ee)

**Comment by liamjdavis** [comment]
Addressed in PR #12.

**Comment by ghost** [comment]
Resolved & Merged! Great work @liamjdavis
